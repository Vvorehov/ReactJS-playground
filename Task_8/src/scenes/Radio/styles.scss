$brandColor: #ED5664;
$secondaryColor: #BDBBBE;

.radio {
  display: inline-block;
  max-width: 100%;
  font-size: 1.4rem;
  > input {
    display: none;
  }
  > span {
    position: relative;
    display: inline-block;
    font-size: inherit;
    padding: .5em 1em;
    line-height: 1;
    border: 0;
    outline: 0;
    border-radius: 2px;
    box-shadow: none;
    text-align: center;
    background-color: #333;
    color: #fff;
    cursor: pointer;
    text-transform: uppercase;
    transition-property: color, background;
    transition-duration: .2s;
    transition-timing-function: ease-out;
    // prevent
    user-select: none;
    appearance: none;
    &:focus {
      outline: 1px solid rgba($brandColor, .5);
    }
    &:hover,
    &:active {
      background-color: $brandColor;
    }
  }
  > input:checked + span {
    background-color: $brandColor;
  }
  > input:disabled + span {
    opacity: .5;
    pointer-events: none;
  }

  &--sort {
    font-size: 1.4rem;
    font-weight: 700;
    > span {
      padding: .5em .5em;
      text-transform: none;
      background-color: transparent;
      color: #000;
      &:hover,
      &:active {
        background-color: transparent;
        color: $brandColor;
      }
    }
    > input:checked + span {
      background-color: transparent;
      color: $brandColor;
    }
  }
}