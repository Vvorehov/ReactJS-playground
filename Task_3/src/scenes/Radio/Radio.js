import React, {Component} from 'react';
import './styles.scss'

class Radio extends Component {

  changeHandler = () => {
    console.log("radio changed");
  };

  render() {
    let className = "radio";
    if (this.props.mod) className += " radio--" + this.props.mod;

    return (
      <label className={className}>
        <input type="radio"
               name={this.props.name}
               defaultChecked={this.props.isChecked}
               onChange={this.changeHandler}
               value={this.props.value}/>
        <span>{this.props.label}</span>
      </label>
    );
  }
}

export default Radio;