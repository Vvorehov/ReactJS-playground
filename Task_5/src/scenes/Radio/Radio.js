import React, {Component} from 'react';
import './styles.scss'

class Radio extends Component {

  render() {
    let className = "radio";
    if (this.props.mod) className += " radio--" + this.props.mod;

    return (
      <label className={className}>
        <input type="radio"
               name={this.props.name}
               data-action-type={this.props.dataActionType}
               defaultChecked={this.props.isChecked}
               onChange={this.props.onChange}
               value={this.props.value}/>
        <span>{this.props.label}</span>
      </label>
    );
  }
}

export default Radio;