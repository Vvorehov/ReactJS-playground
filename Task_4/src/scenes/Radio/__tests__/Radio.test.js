import React from "react";
import Radio from "../Radio";
import TestRenderer from 'react-test-renderer';

describe('Radio layout complete', () => {
    test('Render correct', () => {

        const component = TestRenderer.create(
            <Radio name="search-by" label="Title" value="title" isChecked={true}/>
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

