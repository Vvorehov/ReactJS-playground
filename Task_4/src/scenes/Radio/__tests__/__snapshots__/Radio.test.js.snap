// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio layout complete Render correct 1`] = `
<label
  className="radio"
>
  <input
    defaultChecked={true}
    name="search-by"
    onChange={[Function]}
    type="radio"
    value="title"
  />
  <span>
    Title
  </span>
</label>
`;
